tests/unit/test_parse_env_file.py

```diff
@@ -0,0 +1,71 @@
+from poethepoet.helpers import parse_env_file
+import pytest
+
+examples = [
+    (
+        """
+    # empty
+    """,
+        {},
+    ),
+    (
+        """
+        # single word values
+        WORD=something
+        WORD_WITH_HASH=some#thing
+        NUMBER=0
+        EMOJI=😃😃
+        DOUBLE_QUOTED_WORD="something"
+        SINGLE_QUOTED_WORD='something'
+        """,
+        {
+            "WORD": "something",
+            "WORD_WITH_HASH": "some#thing",
+            "NUMBER": "0",
+            "EMOJI": "😃😃",
+            "DOUBLE_QUOTED_WORD": "something",
+            "SINGLE_QUOTED_WORD": "something",
+        },
+    ),
+    (
+        """
+        # multiword values
+        WORD=some\\ thing                   # and trailing comments
+        DOUBLE_QUOTED_WORD="some thing"
+        SINGLE_QUOTED_WORD='some thing'
+        """,
+        {
+            "WORD": r"some thing",
+            "DOUBLE_QUOTED_WORD": "some thing",
+            "SINGLE_QUOTED_WORD": "some thing",
+        },
+    ),
+    (
+        """
+        # values with line breaks
+        WORD=some\\
+thing
+        DOUBLE_QUOTED_WORD="some
+        thing"
+        SINGLE_QUOTED_WORD='some
+        thing'
+        """,
+        {
+            "WORD": "some\nthing",
+            "DOUBLE_QUOTED_WORD": "some\n        thing",
+            "SINGLE_QUOTED_WORD": "some\n        thing",
+        },
+    ),
+    (
+        """
+    # without linebreak between vars
+    FOO=BAR BAR=FOO
+    """,
+        {"FOO": "BAR", "BAR": "FOO"},
+    ),
+]
```

Comment:
Can we add a test for space before the `FOO`, too? And, instead of strictly complying, why not allow spaces between the `=`?

Reply:
Look again, preceding space is present in all the tests 💯  Maybe there should be a test *without* preceding space...

I considered allowing spaces around the `=` though felt that sticking to how bash does it was a safer bet design wise. I figured it would be least surprising if the file is loaded exactly the same way as it would be via the source command. Maybe this is worth revisiting, though I would need evidence to motivate a change.